pkg/providers/filesystem.go

```diff
@@ -0,0 +1,134 @@
+package providers
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/karrick/godirwalk"
+	"github.com/spectralops/teller/pkg/core"
+
+	"github.com/spectralops/teller/pkg/logging"
+)
+
+type FileSystem struct {
+	logger        logging.Logger
+	rootDirectory string
+}
+
+// NewFileSystem creates new provider instance
+func NewFileSystem(logger logging.Logger) (core.Provider, error) {
+
+	return &FileSystem{
+		logger:        logger,
+		rootDirectory: "",
+	}, nil
+}
+
+// Name return the provider name
+func (f *FileSystem) Name() string {
+	return "FileSystem"
+}
+
+// Put will create a new single entry
+func (f *FileSystem) Put(p core.KeyPath, val string) error {
+	return f.writeFile(f.getFilePath(p.Path), val)
+}
+
+// PutMapping will create a multiple entries
+func (f *FileSystem) PutMapping(p core.KeyPath, m map[string]string) error {
+	for k, v := range m {
+		ap := p.SwitchPath(fmt.Sprintf("%v/%v", f.getFilePath(p.Path), k))
+		err := f.Put(ap, v)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// GetMapping returns a multiple entries
+func (f *FileSystem) GetMapping(p core.KeyPath) ([]core.EnvEntry, error) {
+
+	findings := []core.EnvEntry{}
+	err := godirwalk.Walk(f.getFilePath(p.Path), &godirwalk.Options{
+		Callback: func(osPathname string, de *godirwalk.Dirent) error {
+			if de.IsRegular() {
+				content, err := f.readFile(osPathname)
+				if err != nil {
+					f.logger.WithError(err).WithField("path", p.Path).Debug("file not found in path")
+					return nil
+				}
+				findings = append(findings, p.FoundWithKey(strings.Replace(osPathname, fmt.Sprintf("%s/", p.Path), "", 1), string(content)))
```

Comment:
At this point, do we have a normalized path? i.e.:

1. All expansions expanded (e.g. `~/foobar`)
2. All path separators work (e.g windows vs unix, `\` vs `/`)

Reply:
Question 1:
the idea is to normalize the path by the configuration. for example:
```yaml
filesystem:
    env_sync:
      path:  /tmp/teller/redis/config
```
and i have `bar` file under the path: `/tmp/teller/redis/config`

the result will be:
```sh
$ go run main.go json
{
  "bar": "bar",
}
```
and not 
```sh
$ go run main.go json
{
  "/tmp/teller/redis/config/bar": "bar",
}
```
make sense?

Reply:
Question 2:
nice catch! changed

Reply:
yes makes sense -- just 1 important factor to take into account:

🚨 users will make typos/mistakes when they write their configuration

so if you take that into account - we're good